sway/tree/container.c

```diff
@@ -1058,6 +1072,75 @@ static void set_fullscreen(struct sway_container *con, bool enable) {
 				con->view->foreign_toplevel, enable);
 		}
 	}
+
+	if (!server.linux_dmabuf_v1 || !con->view->surface) {
+		return;
+	}
+	if (!enable) {
+		wlr_linux_dmabuf_v1_set_surface_feedback(server.linux_dmabuf_v1,
+			con->view->surface, NULL);
+		return;
+	}
+
+	if (!con->pending.workspace || !con->pending.workspace->output) {
+		return;
+	}
+
+	struct sway_output *output = con->pending.workspace->output;
+	struct wlr_output *wlr_output = output->wlr_output;
+
+	// TODO: add wlroots helpers for all of this stuff
+
+	const struct wlr_drm_format_set *renderer_formats =
+		wlr_renderer_get_dmabuf_texture_formats(server.renderer);
+	assert(renderer_formats);
+
+	int renderer_drm_fd = wlr_renderer_get_drm_fd(server.renderer);
+	int backend_drm_fd = wlr_backend_get_drm_fd(wlr_output->backend);
```

Comment:
Aren't those the same?

Reply:
Not necessarily:

- The renderer FD points to the DRM render node, the backend FD points to the primary node.
- On split render/display setups, backend FD points to the display-only device, renderer FD points to the render-only FD (well… not yet, but soon)
- When the user overrides the rendering device (`WLR_RENDER_DRM_DEVICE`), then they might not match.
- When we'll support multi-GPU scanout (https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3331), they might not match.

Reply:
Ah yes, the infamous split render/display setup, forgot about that one.